46.md

```diff
@@ -1,98 +1,162 @@
+
+The remote signer would provide a connection token in the form:
 
 ```
-bunker://<hex-pubkey>?relay=wss://...&relay=wss://...&secret=<optional-secret>
+bunker://<remote-pubkey>?relay=<wss://relay-to-connect-on>&relay=<wss://another-relay-to-connect-on>&secret=<optional-secret-value>
 ```
 
-The user copies that token and pastes it in the client UI somehow. Then the client can send events of kind `24133` to the specified relays and wait for responses from the remote signer.
+This token is pasted into the client by the user and the client then uses the details to connect to the remote signer via the specified relay(s).
 
-### Started by the client
+### Direct connection initiated by the client
 
-The client generates a QR code in the following form (URL-encoded):
+In this case, basically the opposite direction of the first case, the client provides a connection token (or encodes the token in a QR code) and the signer initiates a connection to the client via the specified relay(s).
 
 ```
-nostrconnect://<client-key-hex>?relay=wss://...&metadata={"name":"...", "url": "...", "description": "..."}
+nostrconnect://<local-keypair-pubkey>?relay=<wss://relay-to-connect-on>&metadata=<json metadata in the form: {"name":"...", "url": "...", "description": "..."}>
 ```
 
-The signer scans the QR code and sends a `connect` message to the client in the specified relays.
+### Remote signer discovery via NIP-89
+
+In this last case, most often used to fascilitate an OAuth-like signin flow, the client first looks for remote signers that have announced themselves via NIP-89 application handler events.
+
+First the client will query for `kind: 31989` events that have a `k` tag of `24133`.
+
+These are generally shown to a user, and once the user selects which remote signer to use and provides the remote pubkey they want to use (via npub, pubkey, or nip-05 value), the client can initiate a connection. Note that it's on the user to select the remote signer that is actually managing the remote key that they would like to use in this case. If the remote pubkey is managed on another remote signer, the connection will fail.
+
+In addition, it's important that clients validate that the pubkey of the announced remote signer matches the pubkey of the `_` entry in the `/.well-known/nostr.json` file of the remote signer's announced domain.
+
+Clients that allow users to create new accounts should also consider validating the availability of a given username in the namespace of remote signer's domain by checking the `/.well-known/nostr.json` file for existing usernames. Clients can then show users feedback in the UI before sending a `create_account` event to the remote signer and receiving an error in return. Ideally, remote signers would also respond with understandable error messages if a client tries to create an account with an existing username.
```

Comment:
This could be removed if my suggestion above is accepted.

Reply:
I think this should be removed or moved to a "best practices"; the checking is a nice-to-have for a better UX, but is in no way required for this flow.

Reply:
I think it should stay with the rest of the "discovery" section so that it's coherent. I've tried to make it clear that this is all a suggestion. I can strengthen the language there.